### ISSUE TITLE
[css-ruby] ruby base container as containing block

### ISSUE THREAD

**Issue by fantasai** [w3c/csswg-drafts#6324]
@TalbotG recently asked about this test from @upsuper  http://wpt.live/css/css-ruby/abs-in-ruby-base-container.html which... I'm not really sure what it's trying to assert.

I don't think our spec is clear what happens when various ruby boxes are relpos'ed, do they form containing blocks for abspos descendants or no, and we should figure out what we should be doing and clarify it.

**Comment by fantasai** [comment]
And probably the tests should be made a little less confusing... relying on cancelling 50px half-leading inside the abspos by using a -1em offset is pretty convoluted. @_@

**Comment by upsuper** [comment]
I don't fully recall all the details of how I came up with this testcase, but I believe it's basically treating the ruby box as normal inline box in the term of abspos children, which I think makes sense.

Also the change originally came from [a compat issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1641139) that someone trying to use abspos in ruby, and Firefox didn't show anything at that time (because I didn't think about it so missed something). This test is basically just checking that abspos in ruby does show up.

**Comment by TalbotG** [comment]
"
If the box has position: absolute:
    The containing block is established by the nearest ancestor box that establishes an absolute positioning containing block, in the following way:
(...)
    If the ancestor is an inline box,
        the containing block is formed by the block-start and inline-start content edges of the first box fragment of the ancestor, and the block-end and inline-end content edges of the last box fragment of the ancestor. 
Note: If the ancestor breaks across a line, the 'start' position might more end-ward than the 'end' position.
"
coming from
[CSS Positioned Layout 3, 
§2.1. Containing Blocks of Positioned Boxes](https://www.w3.org/TR/css-position-3/#def-cb)
and so, that includes cases where the inline element is split across multiple lines.

I have old CSS2.x tests on this:
http://www.gtalbot.org/BrowserBugsSection/css21testsuite/containing-block-017-GT.html
http://www.gtalbot.org/BrowserBugsSection/css21testsuite/containing-block-017-GT2.html
http://www.gtalbot.org/BrowserBugsSection/css21testsuite/containing-block-031.html
http://www.gtalbot.org/BrowserBugsSection/css21testsuite/containing-block-032.html

I remember reporting that some of tests in the CSS2.1 test suite regarding an abs. pos. inline inside a rel. pos. inline had various problems:
https://lists.w3.org/Archives/Public/public-css-testsuite/2012Mar/0008.html
Other tests may be weak:
http://test.csswg.org/suites/css21_dev/20110323/html4/containing-block-013.htm

**Comment by TalbotG** [comment]
What I am trying to say is this: an absolutely positioned inline inside a relatively positioned inline is a complex code scenario and current browsers (Firefox, Chromium, Safari) do not implement, do not comply with past and current spec. We have a few tests, most of them are weak, I'd say. 
The only browser which was very good was Operan 12.16 (Presto engine).

**Comment by TalbotG** [comment]
> Firefox didn't show anything at that time (because I didn't think about it so missed something). This test is basically just checking that abspos in ruby does show up.

Indeed, Firefox 78.11.0 ESR does not show the <span class="abs">X</span> but Firefox 91.0a1 buildID=20210625093436 does.

Okay. I can not approve or disapprove your abs-in-ruby-base-container.html test, Xidorn.

I know that the reference could be made much more simple and direct with no abs. or rel. pos. and no offset.

**Comment by fantasai** [comment]
See also https://github.com/web-platform-tests/wpt/pull/29482
